### ISSUE TITLE
.babelrc not working as expected. 

### ISSUE THREAD

**Issue by oliverbenns** [gatsbyjs/gatsby#597]
The [docs state](https://github.com/gatsbyjs/gatsby#configuring-babel) that you can configure your own `.babelrc` and use any plugins that babel already depends on. However if I create a `.babelrc` file in the project root with the following config, it fails to use the already packed in `transform-object-assign`:

```
{
  "presets": [
    'react',
    'es2015',
    'stage-0'
  ],
  "plugins": [
    'add-module-exports',
    'transform-object-assign'
  ]
}
```

This results in the `gatsby-example-site` failing to run in IE (tested on 10) because `Object.assign` is never transformed and is being called in the `react-responsive-grid` dependency.

[Example repo](https://github.com/oliverbenns/babelrc-gatsby-test)

**Comment by KyleAMathews** [comment]
Hmmm... so after poking around some I confirmed our [babelrc loader](https://github.com/gatsbyjs/gatsby/blob/master/lib/utils/babel-config.js) is working as expected. My guess is that some lib or component the default starter is loading has an Object.assign in it which is breaking things. Gatsby's webpack config for Babel is setup to only transform *your* code not code from `node_modules`. 

The [plugin](https://github.com/babel/babel/tree/master/packages/babel-plugin-transform-object-assign) actually recommends using a polyfill for Object.assign and only using this transform for libraries — perhaps as with Gatsby, most people don't want to transform 3rd party code.

I've used this polyfill before https://www.npmjs.com/package/es6-object-assign
